packages/features/ee/workflows/api/scheduleEmailReminders.ts

```diff
@@ -40,10 +70,11 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
   // delete batch_ids with already past scheduled date from scheduled_sends
   const pageSize = 90;
   let pageNumber = 0;
-  const deletePromises: Promise<any>[] = [];
+
+  const remindersToDelete: { referenceId: string | null }[] = [];
 
   while (true) {
-    const remindersToDelete = await prisma.workflowReminder.findMany({
+    const newRemindersToDelete = await prisma.workflowReminder.findMany({
```

Comment:
This logic to get remindersToDelete can be abstracted out in a separate function now easily as it's just fetching workflowReminders page by page till it finds no more.
`getAllRemindersToDelete` could be the name.